src/ensRegistry.ts

```
@@ -61,14 +66,19 @@ function _handleNewOwner(event: NewOwnerEvent, isMigrated: boolean): void {
       domain.labelName = label
     }
 
-    if(label == null) {
+    if(label === null) {
       label = '[' + event.params.label.toHexString().slice(2) + ']'
     }
     if(event.params.node.toHexString() == '0x0000000000000000000000000000000000000000000000000000000000000000') {
       domain.name = label
     } else {
       let parent = Domain.load(event.params.node.toHexString())
-      domain.name = label + '.' + parent.name
+      if(parent){
```

This should always be true. Can you throw an error if it's not?